src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -36,117 +32,137 @@
+    private GreengrassV2DataClient client;
     @TempDir
-    Path tmpPath;
-
-    private CertificateManager certificateManager;
-    private CertificateStore certificateStore;
+    Path rootDir;
+    private Kernel kernel;
 
 
     @BeforeEach
-    void beforeEach() {
-        DomainEvents events = new DomainEvents();
-        certificateStore = new CertificateStore(tmpPath, events);
+    void setup(ExtensionContext context) throws DeviceConfigurationException {
+        ignoreExceptionOfType(context, SpoolerStoreException.class);
+
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+        kernel.getContext().put(GroupManager.class, groupManager);
+        kernel.getContext().put(SecurityService.class, securityServiceMock);
+        kernel.getContext().put(CertificateExpiryMonitor.class, certExpiryMonitor);
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+
+        lenient().when(clientFactory.fetchGreengrassV2DataClient()).thenReturn(client);
+    }
 
-        certificateManager = new CertificateManager(
-                certificateStore, mockConnectivityInformation,
-                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), clientFactoryMock, securityServiceMock);
+    @AfterEach
+    void cleanup() {
+        kernel.shutdown();
+    }
 
+    // TODO: Consolidate this test helpers with ClientDevicesAuthServiceTest
+    private void startNucleus() throws InterruptedException {
+        CountDownLatch authServiceRunning = new CountDownLatch(1);
+        Path resourceDirectory = Paths.get(
+                "src", "test", "resources", "com", "aws", "greengrass", "clientdevices", "auth", "config.yaml");
+        Path testPath = rootDir.toAbsolutePath();
+        kernel.parseArgs("-r", testPath.toString(), "-i", resourceDirectory.toFile().getAbsolutePath());
+        kernel.getContext().addGlobalStateChangeListener((service, was, newState) -> {
+            if (ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME.equals(service.getName()) && service.getState()
+                    .equals(State.RUNNING)) {
+                authServiceRunning.countDown();
+            }
+        });
+        kernel.launch();
+        assertThat(authServiceRunning.await(30L, TimeUnit.SECONDS), is(true));
     }
 
-    /**
-     * Simulates an external party configuring their custom CA.
-     */
-    private X509Certificate[] arrangeCustomCertificateAuthority() throws NoSuchAlgorithmException, CertificateException,
-            OperatorCreationException, CertIOException, URISyntaxException, KeyLoadingException,
-            ServiceUnavailableException, CertificateChainLoadingException, InvalidConfigurationException {
-        // Generate custom certificates
+    private static Pair<X509Certificate[], KeyPair[]> arrangeCredentials() throws NoSuchAlgorithmException,
```

Comment:
I know you're basing this naming off that article you linked, but `arrangeCredentials` doesn't mean anything to me. IMO it's just a *really* terrible name. Can you make it clear what's actually happening? Rename to `generateCoreAndIntermediateCA` or something